src/sarif_converter.ts

```diff
@@ -0,0 +1,180 @@
+import { IAnalysisResult, ISuggestion, IFileSuggestion } from './interfaces/analysis-result.interface';
+import { Log, ReportingConfiguration, ReportingDescriptor  } from "sarif"
+
+interface ISarifSuggestion extends IFileSuggestion {
+  id: string;
+  ruleIndex: number;
+  rule: ReportingDescriptor 
+  level: ReportingConfiguration.level;
+  text: string;
+  file: string;
+}
+interface ISarifSuggestions {
+  [suggestionIndex: number]: ISarifSuggestion;
+}
+class Sarif {
+  private analysisResults: IAnalysisResult;
+  private suggestions: ISarifSuggestions= {}
+
+  constructor(analysisResults: IAnalysisResult) {
+    this.analysisResults = analysisResults;
+    for (const [file] of Object.entries(this.analysisResults.files)) {
+      for (const [issueId, issue] of <[string, IFileSuggestion][]>Object.entries(this.analysisResults.files[file])) {
+        if (!this.suggestions || !Object.keys(this.suggestions).includes(issueId)) {
+          this.suggestions[issueId] = { ...issue[0], file: file.substring(1) };
```

Comment:
Why do you need this step? Can't you just execute this code in the sarifConverter by passing the analysisResults as a parameter?